tests/System.Slices.Tests/MemoryTests.cs

```diff
@@ -30,25 +31,72 @@ public void SimpleTestS()
         [Fact]
         public void ArrayMemoryLifetime()
         {
-            Memory<byte> copyStoredForLater;
-            using (var owner = new OwnedArray<byte>(1024)) {
-                Memory<byte> memory = owner.Memory;
-                Memory<byte> memorySlice = memory.Slice(10);
-                copyStoredForLater = memorySlice;
-                using (memorySlice.Reserve()) { // increments the "outstanding span" refcount
-                    Assert.Throws<InvalidOperationException>(() => { // memory is reserved; cannot dispose
-                        owner.Dispose();
-                    });
-                    Span<byte> span = memorySlice.Span;
-                    span[0] = 255;
-                } // releases the refcount
-            }
+            Memory<byte> copyStoredForLater = default(Memory<byte>);
+            Assert.Throws<InvalidOperationException>(() => {
```

Comment:
We could make it so on the first call to Dispose, it calls Dispose(true), and after that Dispose(false) until it is reinitialized.  So it isn't an exception, but can be programmatically used.